Initial version 0.1.0 changes from original repository

mdeinum opened the pull request:
Development of Spring Batch Excel continued in the original repository at
http://github.com/mdeinum/spring-batch-excel.git, with this commit we transfer all fixes, updates and new features in one go to the Spring Batch Excel repository.

The most important changes are:
- Moved to new location org.springframework.batch.extensions
- Removed support for JXL
- Initial support for reading XLSX files streaming
- Upgraded dependencies to most recent versions
- Support for reading password protected files
- Numerous small fixes

Fixes: #10, #11, #19, #26, #29, #34
Closes: #13, #14, #30, #35, #50

benas commented:
This is awesome! Thank you for all these updates Marten! Please make sure to update the date in license headers (I saw a couple of 2011 and 2019) and add your name in the `@author` tag of Javadoc. Other than that, the PR LGTM and could be merged 👍

mdeinum commented:
I went through the files and added the missing headers author and since tags. I also applied the spring format plugin which showed me some inconsistencies in using `this`, I also removed the last compiler warnings I was having on the main code. 

So if you could have a final look and we agree this is fine, then lets move it to the main branches and see if we can get a release out.

benas commented:
Great! LGTM as a first iteration. Rebased and merged.

> Fixes: #10, #11, #19, #26, #29, #34
> Closes: #13, #14, #30, #35, #50

@mdeinum Merging the PR did not automatically close those issues. Please go ahead a close them with appropriate comments if needed. I will create an issue for the first release. Thank you!
